### DISCUSSION TITLE
Sharing the Wayland work

### DISCUSSION THREAD

**Discussion by jrseliga** [regolith-linux/regolith-desktop#964] [Ideas]
The tightly coupled integration that Regolith provides between Gnome and a tiling window manager has always been intriguing to me.

I know this topic is out of place for the bigger goals of this project, but I've been watching the Wayland support work since the [GSoC'22](https://regolith-desktop.com/posts/sway_integration_gsoc/).

I've come to appreciate Fedora a lot recently and started thinking that a lot if not all of that effort put into Regolith to support this integration could be usable by the wider community.

I'm curious what this community's and project contributor's thoughts are on this generally. But also specifically around things like:
- Could the various packages like `regolith-displayd` or `regolith-session` be bundled for common package managers (`apt`, `yum`, `dnf`, etc.)?
- Could the packages that target i3/Sway explicitly be updated to easily support other tiling managers

**Comment by kgilmer** [comment]
Hi @jrseliga ,

> I've come to appreciate Fedora a lot recently and started thinking that a lot if not all of that effort put into Regolith to support this integration could be usable by the wider community.

Yeah, producing packages suitable for Fedora has been a long time request.  I have looked into it a few times briefly but have yet to be sufficiently freed up to begin serious work on it.

> Could the various packages like regolith-displayd or regolith-session be bundled for common package managers (apt, yum, dnf, etc.)?

I believe so, yes.  We need a way of staging packaging metadata somewhere (for Debian packages we just put the metadata right in the source repo, due to the origins of the project being started from Ubuntu), and updating the [build system](https://github.com/regolith-linux/voulage/blob/main/.github/scripts/ext-rpm.sh) to build produce the packages in a way that is idiomatic for the distribution.

> Could the packages that target i3/Sway explicitly be updated to easily support other tiling managers

Well, I guess it depends on your definition of "easily" but yes, the package `regolith-wm-config` already supports two window managers.  There is a namespace convention ([example](https://github.com/regolith-linux/regolith-wm-config/blob/main/usr/share/regolith/sway/config.d/55_session_keybindings#L40)) such that variances in window managers can be scoped to a namespace.

Some features that Regolith utilizes from the underlying package manager, that would be needed for "easy" porting:

* support for "virtual" or abstract packages of which one or more concrete packages can satisfy
* support for soft and hard dependencies (installed by default, but removable without causing rdepend package from being removed)
* support to express conflicts across packages
